pkgs/by-name/pr/protonmail-desktop/package.nix

```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchurl, wrapGAppsHook, makeWrapper, autoPatchelfHook, pkgs
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, dpkg
+, glib
+, gtk3
+, libX11
+, libXtst
+, libdrm
+, mesa
+, nss
+, udev
+, xdg-utils
+, coreutils
+, xorg
+
+# command line arguments which are always set e.g "--disable-gpu"
+, commandLineArgs ? ""
+
+# Necessary for USB audio devices.
+, pulseSupport ? stdenv.isLinux
+, libpulseaudio
+
+# For GPU acceleration support on Wayland (without the lib it doesn't seem to work)
+, libGL
+
+# For video acceleration via VA-API (--enable-features=VaapiVideoDecoder,VaapiVideoEncoder)
+, libvaSupport ? stdenv.isLinux
+, libva
+}:
+
+let
+  inherit (lib) optional makeLibraryPath makeSearchPathOutput makeBinPath;
+
+  deps = [
+    alsa-lib at-spi2-atk at-spi2-core atk
+    glib gtk3 libdrm libX11 libGL
+    libXtst mesa nss udev xorg.libxcb
+  ]
+    ++ optional pulseSupport libpulseaudio
+    ++ optional libvaSupport libva;
+
+  rpath = makeLibraryPath deps + ":" + makeSearchPathOutput "lib" "lib64" deps;
+  binpath = makeBinPath deps;
+in
+stdenv.mkDerivation rec {
+  pname = "ProtonMail-desktop";
+  baseUrl = "https://proton.me/download/mail/linux/";
+  version = "beta";
+
+  src = fetchurl {
+    url = "${baseUrl}/${pname}-${version}.deb";
+    sha256 = "sha256-3FRm02Ew9GuGMXFAkwSL+FbeLyaI1SmpDU3A5eJh6TU=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    (wrapGAppsHook.override { inherit makeWrapper; })
+  ];
+
+  buildInputs = with pkgs; [ gtk3 libnotify libdrm gdb nss mesa alsa-lib ];
+
+  unpackPhase = "dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner";
+
+  installPhase = ''
+    mkdir -p $out
+    cp -av usr/* $out
+    rm -rf usr
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # Add command line args to wrapGApp.
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH : ${rpath}
+      --prefix PATH : ${binpath}
+      --suffix PATH : ${lib.makeBinPath [ xdg-utils coreutils ]}
+    )
+  '';
+
+  meta = with lib; {
+    description = "Desktop application for Proton Mail";
+    homepage = "https://www.proton.me/mail";
+    license = licenses.free;
```

Comment:
Which one?

Reply:
GPLv3 maybe? https://github.com/ProtonMail/inbox-desktop
But that repo doesn't actually have the source code

Reply:
Hey, from what I know the license is MIT :)

Reply:
Can you send a link confirming that?

Reply:
this one?: https://github.com/ProtonMail/inbox-desktop/blob/main/LICENSE

Reply:
No. Something can't be licensed under GPL without the code being public.

Reply:
iiuc, the code does not need to be public, you just need the ability to request it

Reply:
I've reached out to proton team with question about license and source code availability.

Reply:
Hey, to maybe clear this up a bit: I'm part/employee of Proton (not that particular team) and already reached out to them to clarify - what I can already say for sure it's planned as with all our clients to open-source the source-code :v:

Reply:
changed to gpl3Plus as this seems to be most accurate. @SebTM in case you get any conflicting info from the team, feel free to update.